ubsan_arangodb_suppressions.txt

```diff
@@ -26,6 +26,7 @@ vptr:s2cell.cc
 vptr:v8::Platform
 # TODO(MBkkt) Some issue in our version of V8, only in arangodbtests
 vptr:v8::internal::Isolate::~Isolate
+vptr:V8PlatformFeature::unprepare()
```

Comment:
What kind of UB is that? `V8PlatformFeature::unprepare()` is our own code - can we fix it instead of suppressing it?

Reply:
> What kind of UB is that? V8PlatformFeature::unprepare() is our own code - can we fix it instead of suppressing it?

```
/usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:85:2: runtime error: member call on address 0x60d00010d280 which does not point to an object of type 'v8::Platform'
0x60d00010d280: note: object has invalid vptr
 00 00 00 00  90 39 c7 a6 cd 55 00 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  00 00 00 00
              ^~~~~~~~~~~~~~~~~~~~~~~
              invalid vptr
    #0 0x55cd9fe2a57a in operator() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:85
    #1 0x55cd9fe2a57a in std::__uniq_ptr_impl<v8::Platform, std::default_delete<v8::Platform> >::reset(v8::Platform*) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:182
    #2 0x55cd9fe27e37 in reset /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:456
    #3 0x55cd9fe27e37 in arangodb::V8PlatformFeature::unprepare() /work/ArangoDB/lib/ApplicationFeatures/V8PlatformFeature.cpp:250
    #4 0x55cda152c026 in arangodb::application_features::ApplicationServer::unprepare() /work/ArangoDB/lib/ApplicationFeatures/ApplicationServer.cpp:796
    #5 0x55cda1518b3d in arangodb::application_features::ApplicationServer::run(int, char**) /work/ArangoDB/lib/ApplicationFeatures/ApplicationServer.cpp:227
    #6 0x55cd974da7c3 in runServer(int, char**, arangodb::ArangoGlobalContext&) /work/ArangoDB/arangod/RestServer/arangod.cpp:156
    #7 0x55cd974d9a49 in main /work/ArangoDB/arangod/RestServer/arangod.cpp:264
    #8 0x7f5233de9d8f in __libc_start_call_main csu/../sysdeps/nptl/libc_start_call_main.h:58:16
    #9 0x7f5233de9e3f in __libc_start_main csu/../csu/libc-start.c:392:3
    #10 0x55cd97419604 in _start (/work/ArangoDB/build/bin/arangod+0x5635604) (BuildId: f313261797f701876f4bdd9bc0bb4395809ee886)

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:85:2 in 
```